DOCKER.md

```diff
@@ -0,0 +1,98 @@
+# Using Docker and Docker Compose for Development and Testing
+
+These instructions are designed for setting up The Rails Port for development and testing using [Docker](https://www.docker.com/). This will allow you to install the OpenStreetMap application and all its dependencies in Docker images and then run them in containers, almost with a single command. You will need to install Docker and Docker Compose on your development machine:
+
+- [Install Docker](https://docs.docker.com/install/)
+- [Install Docker Compose](https://docs.docker.com/compose/install/)
+
+The first step is to fork/clone the repo to your local machine:
+
+    git clone https://github.com/openstreetmap/openstreetmap-website.git
+
+Now change working directory to the `openstreetmap-website`:
+
+    cd openstreetmap-website
+
+## Initial Setup
+
+### Storage
+
+    cp config/example.storage.yml config/storage.yml
+
+### Database
+
+    cp config/docker.database.yml config/database.yml
+
+## Prepare local settings file
+
+This is a workaround. [See issues/2185 for details](https://github.com/openstreetmap/openstreetmap-website/issues/2185#issuecomment-508676026).
+
+    touch config/settings.local.yml
+
+## Installation
+
+To build local Docker images run from the root directory of the repository:
+
+    docker-compose build
+
+If this is your first time running or you have removed cache this will take some time to complete. Once the Docker images have finished building you can launch the images as containers.
+
+To launch the app run:
+
+    docker-compose up -d
+
+This will launch one Docker container for each 'service' specified in `docker-compose.yml` and run them in the background. There are two options for inspecting the logs of these running containers:
+
+- You can tail logs of a running container with a command like this: `docker-compose logs -f web` or `docker-compose logs -f db`.
+- Instead of running the containers in the background with the `-d` flag, you can launch the containers in the foreground with `docker-compose up`. The downside of this is that the logs of all the 'services' defined in `docker-compose.yml` will be intermingled. If you don't want this you can mix and match - for example, you can run the database in background with `docker-compose up -d db` and then run the Rails app in the foreground via `docker-compose up web`.
+
+### Migrations
+
+Run the Rails database migrations:
+
+    docker-compose run --rm web bundle exec rake db:migrate
+
+### Tests
+
+Run the test suite by running:
+
+    docker-compose run --rm web bundle exec rake test:db
+
+### Loading an OSM extract
```

Comment:
Since loading an extract via osmosis stopped working after PR https://github.com/openstreetmap/openstreetmap-website/pull/2432 I'm inclined to remove this section until I can offer a working solution.

Reply:
There’s a work songs available: https://github.com/openstreetmap/openstreetmap-website/issues/2543#issuecomment-587889501

Just add it to the description

Reply:
It’s a super-ugly workaround, almost to the point that I think it’d be harmful just to recommend that people monkey with the schema pre- and post-import. Let’s just remove the section for now?

Reply:
It’s as ugly as having to tweak sequences. My point was, it was already ugly before, we’re not adding much to that by this workaround.

Reply:
Osmosis 0.47.1 now correctly writes to the DB so this section should work fine.